File: src/main/java/com/github/trcdeveloppers/clayium/blocks/ores/BlockDenseClayOre.java
package com.github.trcdeveloppers.clayium.blocks.ores;


import com.github.trcdeveloppers.clayium.annotation.CBlock;
import com.github.trcdeveloppers.clayium.blocks.ClayiumBlocks;
import com.github.trcdeveloppers.clayium.items.ItemClayShovel;
import net.minecraft.block.Block;
import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;

import javax.annotation.ParametersAreNonnullByDefault;

import static com.github.trcdeveloppers.clayium.creativetab.ClayiumCreativeTab.CLAYIUM;

@CBlock(registryName = "dense_clay_ore")
public class BlockDenseClayOre extends Block implements ClayiumBlocks.ClayiumBlock {
    public BlockDenseClayOre(Material material) {
        super(material);
        this.setCreativeTab(CLAYIUM);
        this.setLightLevel(0f);
        this.setHarvestLevel("pickaxe", 1);
        this.setHardness(3f);
        this.setResistance(5f);
        this.setSoundType(SoundType.STONE);
    }

    @SuppressWarnings("unused")
    public BlockDenseClayOre() {
        this(Material.ROCK);
    }

    @Override
    @ParametersAreNonnullByDefault
    public boolean canHarvestBlock(IBlockAccess world, BlockPos pos, EntityPlayer player) {
        if (player.getHeldItemMainhand().getItem() instanceof ItemClayShovel) return true;
        return super.canHarvestBlock(world, pos, player);
    }
}

File: src/main/kotlin/com/github/trcdevelopers/clayium/common/blocks/ores/BlockDenseClayOre.kt
package com.github.trcdevelopers.clayium.common.blocks.ores

import com.github.trcdevelopers.clayium.common.items.ItemClayShovel
import net.minecraft.block.Block
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess

class BlockDenseClayOre : Block(Material.ROCK) {
    init {
        setSoundType(SoundType.STONE)
        setLightLevel(0f)
        setResistance(5f)
        setHardness(3f)
        setHarvestLevel("pickaxe", 1)
    }

    override fun canHarvestBlock(world: IBlockAccess, pos: BlockPos, player: EntityPlayer): Boolean {
        return if (player.heldItemMainhand.item is ItemClayShovel) true else super.canHarvestBlock(world, pos, player)
    }
}
